store/store.go

```diff
@@ -1174,6 +1174,25 @@ func (s *Store) Query(qr *proto.QueryRequest) ([]*proto.QueryRows, error) {
 	if qr.Level == proto.QueryRequest_QUERY_REQUEST_LEVEL_NONE && s.isStaleRead(qr.Freshness, qr.FreshnessStrict) {
 		return nil, ErrStaleRead
 	}
+
+	if qr.LeaderReadOpt {
+		return s.leaderRead(qr)
```

Comment:
This is not quite what we should do. It should read like this:
```Go
if qr.LeaderReadOpt() {
    if err := s.VerifyLeader(); err != nil {
        return nil, err
   }
}
return s.db.Query(qr.Request, qr.Timings)
```

We then delete `s.leaderRead()` and add `s.VerifyLeader()`
```Go
func (s *Store) VerifyLeader() err {
    return future := s.raft.VerifyLeader().Error()
}
```
This also has the advantage of exposing `VerifyLeader()`, making it easy to unit test.